composer/loggers/remote_uploader_downloader.py

```diff
@@ -163,16 +187,24 @@ class RemoteUploaderDownloader(LoggerDestination):
     """
 
     def __init__(self,
-                 object_store_cls: Type[ObjectStore],
-                 object_store_kwargs: Dict[str, Any],
-                 object_name: str = '{remote_file_name}',
+                 bucket_uri: str,
+                 backend_kwargs: Optional[Dict[str, Any]] = None,
+                 path_format_string: str = '{remote_file_name}',
                  num_concurrent_uploads: int = 4,
                  upload_staging_folder: Optional[str] = None,
                  use_procs: bool = True,
                  num_attempts: int = 3) -> None:
-        self.object_store_cls = object_store_cls
-        self.object_store_kwargs = object_store_kwargs
-        self.object_name = object_name
+        parsed_remote_bucket = urlparse(bucket_uri)
+        self.remote_backend_name, remote_bucket_name = parsed_remote_bucket.scheme, parsed_remote_bucket.netloc
+        self.backend_kwargs = backend_kwargs if backend_kwargs is not None else {}
+        if self.remote_backend_name == 's3' and 'bucket' not in self.backend_kwargs:
+            self.backend_kwargs['bucket'] = remote_bucket_name
+        elif self.remote_backend_name == 'sftp' and 'host' not in self.backend_kwargs:
+            self.backend_kwargs['host'] = f'sftp://{remote_bucket_name}'
+        elif self.remote_backend_name == 'libcloud' and 'container' not in self.backend_kwargs:
```

Comment:
if the user does not provide a `provider`, then we just punt the error to the `LibCloudLogger` error?

Reply:
Yes. the intention of this conditional block was to grab as much as possible automatically from the uri. Anything else that is required will raise an error when it tries to create the relevant object store class and is missing a required param.